### PR TITLE
CM Reconcile +++ env vars merge

### PR DESCRIPTION
**Pull request by firegloves**

### CODE REVIEW

File: controllers/reconciliation/component_manager.go [hunk 1]
```diff
@@ -2,12 +2,53 @@ package reconciliation
...
 
+	appsv1 "k8s.io/api/apps/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+const componentManagerDeploymentEndName = "cm-deployment"
+
 func (r *ReconcileManager) reconcileComponentManager(ctx context.Context, image string, req ctrl.Request) error {
-	time.Sleep(2 * time.Second)
+	r.Log.Info("Starting ComponetManager reconciliation flow")
```
Comment (zonia3000) [inline controllers/reconciliation/component_manager.go]: ComponetManager -> ComponentManager

File: controllers/reconciliation/component_manager.go [hunk 2]
```diff
@@ -2,12 +2,53 @@ package reconciliation
...
+	var deployment *appsv1.Deployment
+	for _, item := range deploymentList.Items {
+		if strings.HasSuffix(item.Name, componentManagerDeploymentEndName) {
+			deployment = &item
+			break
+		}
+	}
+
+	if deployment == nil {
+		return fmt.Errorf("deployment ComponetManager not found")
```
Comment (zonia3000) [inline controllers/reconciliation/component_manager.go]: ComponetManager -> ComponentManager

File: controllers/reconciliation/component_manager.go [hunk 3]
```diff
@@ -2,12 +2,53 @@ package reconciliation
...
+	}
+
+	envVars := utils.MergeEnvVars(entandoAppV2, deployment)
+	deployment.Spec.Template.Spec.Containers[0].Env = envVars
+
+	if err := r.Update(ctx, deployment); err != nil {
+		return err
+	}
+
+	r.Log.Info("Finished ComponetManager reconciliation flow")
```
Comment (zonia3000) [inline controllers/reconciliation/component_manager.go]: ComponetManager -> ComponentManager

File: controllers/reconciliation/component_manager.go (hunk 1 above)
Comment (entgigi) [reply controllers/reconciliation/component_manager.go]: done ,tnx

File: controllers/reconciliation/component_manager.go (hunk 2 above)
Comment (entgigi) [reply controllers/reconciliation/component_manager.go]: done ,tnx

File: controllers/reconciliation/component_manager.go (hunk 3 above)
Comment (entgigi) [reply controllers/reconciliation/component_manager.go]: done ,tnx

**Review by zonia3000 — APPROVED** [review]

**Pull request merged**